commit d0f1f569e27f7f34f3bfc03bd759f28f7b567997
Author: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@nokia.com>
Date:   Mon Aug 30 17:16:49 2010 +0200

    Fix QStaticText with OpenGL1 engine

    Like OpenGL2 and OpenVG, the OpenGL1 paint engine also transforms text 
    coordinates itself and does not require QStaticText to use device 
    coordinates.

    Task-number: QTBUG-13228 
    Reviewed-by: Samuel

--- a/src/gui/painting/qpainter.cpp
+++ b/src/gui/painting/qpainter.cpp
@@ -5853,10 +5853,14 @@ void QPainter::drawStaticText(const QPoi
         return;
     }
 
-    if (d->extended->type() == QPaintEngine::OpenGL2 && !staticText_d->untransformedCoordinates) {
+    bool paintEngineSupportsTransformations = d->extended->type() == QPaintEngine::OpenGL2
+                                           || d->extended->type() == QPaintEngine::OpenVG
+                                           || d->extended->type() == QPaintEngine::OpenGL;
+
+    if (paintEngineSupportsTransformations && !staticText_d->untransformedCoordinates) {
         staticText_d->untransformedCoordinates = true;
         staticText_d->needsRelayout = true;
-    } else if (d->extended->type() != QPaintEngine::OpenGL2 && staticText_d->untransformedCoordinates) {
+    } else if (!paintEngineSupportsTransformations && staticText_d->untransformedCoordinates) {
         staticText_d->untransformedCoordinates = false;
         staticText_d->needsRelayout = true;
     }
